src/main/java/com/exactpro/epfast/template/xml/DictionaryFieldOperatorXml.java

```diff
@@ -16,32 +16,40 @@
 
 package com.exactpro.epfast.template.xml;
 
-import com.exactpro.epfast.template.Dictionary;
-
-import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.bind.annotation.XmlAttribute;
 
 public class DictionaryFieldOperatorXml extends FieldOperatorXml {
 
-    private Dictionary dictionary;
-
     private ReferenceImpl dictionaryKey;
 
-    public Dictionary getDictionary() {
-        return dictionary;
-    }
+    private String dictionaryKeyName = "";
 
-    @XmlElement(name = "dictionary", namespace = NamespaceProvider.XML_NAMESPACE)
-    public void setDictionary(Dictionary dictionary) {
-        this.dictionary = dictionary;
+    private String dictionaryKeyNs;
+
+    @XmlAttribute(name = "dictionary")
+    public void setDictionaryName(String dictionary) {
+        super.setDictionaryName(dictionary);
     }
 
     public ReferenceImpl getDictionaryKey() {
         return dictionaryKey;
     }
 
-    @XmlElement(name = "nsKey", namespace = NamespaceProvider.XML_NAMESPACE)
-    public void setDictionaryKey(ReferenceImpl dictionaryKey) {
-        this.dictionaryKey = dictionaryKey;
+    @XmlAttribute(name = "keyName")
+    public void setDictionaryKeyName(String keyName) {
+        this.dictionaryKeyName = keyName;
     }
-}
 
+    @XmlAttribute(name = "keyNs")
+    public void setDictionaryKeyNs(String keyNs) {
+        this.dictionaryKeyNs = keyNs;
+    }
+
+    @Override
+    protected void afterUnmarshal(Unmarshaller unmarshaller, Object parent) {
+        super.afterUnmarshal(unmarshaller, parent);
+        dictionaryKey = new ReferenceImpl(dictionaryKeyName, dictionaryKeyNs);
```

Comment:
Try to implement `ReferenceImpl` the same way as Identity is implemented: pass namespace provider in constructor and delegate setKey/setNs calls to the corresponding methods of `ReferenceImpl`